tests/functional/test_install.py

```diff
@@ -1666,14 +1667,6 @@ def test_install_from_test_pypi_with_ext_url_dep_is_blocked(script, index):
     assert error_cause in res.stderr, str(res)
 
 
-def test_installing_scripts_outside_path_prints_warning(script):
```

Comment:
We should rewrite this test, to not use prefix but still perform this check.

Reply:
Or use `monkeypatch` to temporarily overwrite `PATH` and trigger the message.

Reply:
I've been playing with this for hours and I don't think it's possible to do the way you described. 

The problem is that `python/pip` live no the same path as the `t1` script being installed. What I've tried so far: `del script.environ["PATH"]` - then the test can't find `pip` / `python -m pip`

Thoughts?

Reply:
Maybe it’s possible to use absolute path to invoke pip? Something like

```python
pip = shutil.which("pip", path=script.environ["PATH"])
del script.environ["PATH"]
result = script.run(pip, ...)
```

Reply:
heh... this cannot be done, the logic inside `message_about_scripts_not_on_PATH` has the following line

https://github.com/pypa/pip/blob/e6414d6db6db37951988f6f2b11ec530ed0b191d/src/pip/_internal/operations/install/wheel.py#L167-L169

So even when I run:

```sh
PATH= PYTHONPATH=src /Users/nchepanov/.pyenv/versions/pip/bin/python -m \
pip install --no-index --find-links file:///Users/nchepanov/dev/github/pip/tests/data/packages script_wheel1
```

The warning won't be printed, which makes me think... there's no longer a way to trigger this warning. Makes me curious why the warning was added in the first place. How do you feel about getting rid of the warning?

Reply:
Strong -1 to removing this warning.

The warning is for non-virtual environments, where PATH is not correctly configured (eg: if you don't check add to PATH when installing via the Windows installer).

Reply:
I don't know much about the Windows installed, but I'd like to point out that there's no longer **any** form of use of `pip install` that would **ever** print this warning, even if the `PATH` is literally empty (the worst type of broken).

I'm ambivalent as to how this will be handled :) Let me know - remove the test, xfail it or something else.

Reply:
Windows has a separate Scripts/ directory for things like pip, idle, and all the other stuff; which is different from the one that python.exe sits in.

This warning is still valid for those cases.

Reply:
xfail the test for now, with a note that --prefix can no longer be used. Someone will fix it at some point in the future. :)

Reply:
Yeah, please don’t remove the message. The warning can still appear in some settings, e.g. `pip install --user` when the user-site script/bin directory is not added to PATH. We can’t use `--user` in tests unfortunately since the tests are run in virtual environments, so `xfail` is probably the best we can do.